### PR TITLE
Update guide, add debugging and configuring markdown.

### PR DESCRIPTION
**Pull request by mrutkows**

### CODE REVIEW

**Comment by mrutkows** [comment]
Closes 490

File: docs/wskdeploy_configuring.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Values supplied on the command line using the ```apihost```, ```auth``` and ```namespace``` flags will override values supplied elsewhere (below).
+
+for example the following flags can be used:
+
+```
+$ wskdeploy --apihost <host> --auth <auth> --namespace <namespace>
+```
+
+Command line is considered highes in precedence since it is an intenional user action to override with the latest values.
```
Comment (pritidesai) [inline docs/wskdeploy_configuring.md]: highes = highest

File: docs/wskdeploy_debugging.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+To enable this trace, set the following environment variable in Bash:
+
+# set to any value > 0
+WSK_CLI_DEBUG=1
+
+## Pay attention to Named error messages
+
+Wskpdeloy uses named errors that describe the type of any error found along with additional values that correspond with an error.
+
+For example, if you have an error in your Manifest's YAML, you may see an
```
Comment (pritidesai) [inline docs/wskdeploy_debugging.md]: an ? ... looks incomplete sentence

File: docs/wskdeploy_debugging.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+All current named errors supported by the utility can be found in the lates ```wskdeployerror.go``` source file:
```
Comment (pritidesai) [inline docs/wskdeploy_debugging.md]: lates = latest

File: docs/wskdeploy_debuugging.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Debugging ```wskdeploy```
```
Comment (pritidesai) [inline docs/wskdeploy_debuugging.md]: duplicate files? there is a docs/wskdeploy_debugging.md in this PR

**Comment by pritidesai** [comment]
LGTM, thanks @mrutkows

**Pull request merged**